src/Microsoft.AspNet.HttpOverrides/HttpMethodOverrideMiddleware.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.HttpOverrides
+{
+    public class HttpMethodOverrideMiddleware
+    {
+        private const string xHttpMethodOverride = "X-Http-Method-Override";
+        private readonly RequestDelegate _next;
+        private readonly string _inputName;
+
+        public HttpMethodOverrideMiddleware(RequestDelegate next, string inputName = null)
+        {
+            _next = next;
+            _inputName = inputName;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (_inputName != null)
+            {
+                var form = await context.Request.ReadFormAsync();
+                var methodType = form[_inputName];
+                context.Request.Method = methodType;
+            }
+            else
+            {
+                var xHttpMethodOverrideValue = context.Request.Headers.Get(xHttpMethodOverride);
```

Comment:
My recollection of this is that this is supposed to override only when the current *actual* method is a POST. At least, that's what we did in MVC and/or Web API before. I believe there are security implications to allowing a GET method to change itself to being a DELETE, etc.

@blowdart